lib/matplotlib/colorbar.py

```diff
@@ -682,66 +782,46 @@ def set_ticks(self, ticks, update_ticks=True):
             to using a default locator.
 
         update_ticks : bool, default: True
-            If True, tick locations are updated immediately.  If False, the
-            user has to call `update_ticks` later to update the ticks.
+            As of 3.5 this has no effect.
 
         """
         if np.iterable(ticks):
             self.locator = ticker.FixedLocator(ticks, nbins=len(ticks))
         else:
             self.locator = ticks
-
-        if update_ticks:
-            self.update_ticks()
+        self._long_axis().set_major_locator(self.locator)
         self.stale = True
 
     def get_ticks(self, minor=False):
         """Return the x ticks as a list of locations."""
-        if self._manual_tick_data_values is None:
-            ax = self.ax
-            long_axis = (
-                ax.yaxis if self.orientation == 'vertical' else ax.xaxis)
-            return long_axis.get_majorticklocs()
-        else:
-            # We made the axes manually, the old way, and the ylim is 0-1,
-            # so the majorticklocs are in those units, not data units.
-            return self._manual_tick_data_values
+        return self._long_axis().get_majorticklocs()
```

Comment:
This is not part of your change, but while in the neighborhood you might want to comment on why get_ticks ignores the "minor" kwarg.  Or, with this PR, should the kwarg be activated?

Reply:
Thanks, I made the minor kwarg work and added a small test...